core/src/main/java/io/parsingdata/metal/data/Slice.java

```diff
@@ -29,22 +29,22 @@
 public class Slice {
 
     public final Source source;
-    public final long offset;
+    public final BigInteger offset;
     public final BigInteger length;
 
-    private Slice(final Source source, final long offset, final BigInteger length) {
+    private Slice(final Source source, final BigInteger offset, final BigInteger length) {
         this.source = checkNotNull(source, "source");
-        this.offset = offset;
+        this.offset = checkNotNull(offset, "offset");
         this.length = checkNotNull(length, "length");
     }
 
-    public static Optional<Slice> createFromSource(final Source source, final long offset, final BigInteger length) {
+    public static Optional<Slice> createFromSource(final Source source, final BigInteger offset, final BigInteger length) {
         if (checkNotNull(length, "length").compareTo(ZERO) < 0 || !checkNotNull(source, "source").isAvailable(offset, length)) { return Optional.empty(); }
         return Optional.of(new Slice(source, offset, length));
     }
 
     public static Slice createFromBytes(final byte[] data) {
-        return new Slice(new ConstantSource(checkNotNull(data, "data")), 0, BigInteger.valueOf(data.length));
+        return new Slice(new ConstantSource(checkNotNull(data, "data")), BigInteger.ZERO, BigInteger.valueOf(data.length));
```

Comment:
Elsewhere you've statically imported `BigInteger.ZERO` and `BigInteger.ONE`. Can you do it everywhere? (Also in throughout the test code.)